nmf/sparse_nmf.py

```diff
@@ -0,0 +1,172 @@
+""" Sparse Nonnegative matrix factorization
```

Comment:
Great. But since we are doing it within scikit-learn, better to make this `nmf` as a module under sklearn, right?  so move it there, add also an `__init__.py` file so Python recognizes it as a submodule